[Release]: 0.2.0

averissimo opened the pull request:
## Release checklist:

- [x] Run `desc::normalize()` to normalize DESCRIPTION file
- [x] Make sure all CI actions are green
- [ ] Build the package locally using the command:R CMD build . which will generate a .tar.gz file necessary for the CRAN submission.
  - [ ] Ceate a git tag called `X.X.X-rcX` with the build
- [ ] Submit the package that was build in the previous step via this form: https://cran.r-project.org/submit.html
- [ ] Address CRAN feedback and repeat the submission to CRAN whenever necessary
- [ ] Create a git tag with the final version set to X.X.X on the main branch

## Changes description

- Updates version to 0.2.0 (`DESCRIPTION` & `NEWS.md`)
- Added related PRs to `NEWS.md` items
- Call `desc::normalize()` to cleanup `DESCRIPTION` (sorts fields and packages in a standard way)
- Removed unused packages from `Suggests`
  - `{here}`, `{mockery}` were not used in `{shiny.telemetry}`
    - _(`{here}` is used inside plumber, but that included in its `renv.lock`)_
  - `{covr}`, `{lintr}`, `{spelling}`, `{rcmdcheck}` were necessary for a Mac OS GH action
    - moved their installation to the setup-r step of the workflow (as per the documentation of the workflow)
  - `{stringr}` was used once and it's easily replaced
- Added [triple dash](https://stackoverflow.com/questions/50788277/why-3-dashes-hyphen-in-yaml-file) prefix to `yaml` files
  - Signals start of the document
- Reduced workflow trigger for `prepare-connect` workflow
  - Trigger only on PRs and push to `main`

jakubnowicki commented on DESCRIPTION:
```diff
@@ -1,7 +1,7 @@
 Type: Package
 Title: 'Shiny' App Usage Telemetry
-Version: 0.1.0.9003
+Version: 0.2.0
```
There is a version mismatch (0.1.1 in NEWS and branch name and 0.2.0 here). I'm okay with using 0.2.0, which includes a new feature - MS SQL Server support, but I'm not forcing it.

averissimo replied:
Agree, resolved! Sorry about the mismatch

jakubnowicki reviewed (commented):
Please check my comment. Besides that, windevel-check spotted this:
```
Found the following (possibly) invalid file URI:
  URI: ./CONTRIBUTING.md
    From: README.md
```
Note that `CONTRIBUTING.md` is in `.Rbuildignore`.

averissimo commented:
@jakubnowicki I've solved your comments and did a bit of house cleaning.

House cleaning:

- Added related PRs to `NEWS.md` items
- Call `desc::normalize()` to cleanup `DESCRIPTION` (sorts fields and packages in a standard way)
- Removed unused packages from `Suggests`
  - `{here}`, `{mockery}` were not used in `{shiny.telemetry}`
    - _(`{here}` is used inside plumber, but that included in its `renv.lock`)_
  - `{covr}`, `{lintr}`, `{spelling}`, `{rcmdcheck}` were necessary for a Mac OS GH action
    - moved their installation to the setup-r step of the workflow (as per the documentation of the workflow)
  - `{stringr}` was used once and it's easily replaced
- Added [triple dash](https://stackoverflow.com/questions/50788277/why-3-dashes-hyphen-in-yaml-file) prefix to `yaml` files
  - Signals start of the document
- Reduced workflow trigger for `prepare-connect` workflow
  - Trigger only on PRs and push to `main`

jakubnowicki commented on .github/workflows/check_with_databases.yml:
```diff
@@ -105,7 +106,12 @@ jobs:
-          extra-packages: local::.
+          extra-packages: |
+            local::.
+            any::rcmdcheck
+            any::spelling
```
Are those needed in `pre-commit` setup? If yes, please add them to the contribution guide.

averissimo replied:
I believe they are not needed there.

All but `rcmdcheck` and `covr` should be used in `precommit` and it doesn't have any hooks for these 2. (spelling has its own hook that installs it internally)

jakubnowicki reviewed (approved):
LGTM
